### PR TITLE
feat: support device name in playwright fixtures

### PR DESCRIPTION
**Pull request by yury-s**
Reference https://github.com/microsoft/playwright-java/issues/1369
Reference https://github.com/microsoft/playwright-java/issues/939

### CODE REVIEW

**Comment by yury-s** [comment]
cc @uchagani

File: playwright/src/main/java/com/microsoft/playwright/junit/Options.java [hunk 1]
```diff
@@ -83,6 +84,15 @@ public Options setBrowserName(String browserName) {
     return this;
   }
 
+  public String getDeviceName() {
```
Comment (dgozman) [inline playwright/src/main/java/com/microsoft/playwright/junit/Options.java:87]: It's unfortunate that we cannot read back the actual values here, only the device name. However, that's better than nothing.

**Review by dgozman — APPROVED** [review]

File: playwright/src/main/java/com/microsoft/playwright/junit/impl/DeviceDescriptor.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+class DeviceDescriptor {
+  public String userAgent;
+  public ViewportSize viewport;
+  public Double deviceScaleFactor;
+  public Boolean isMobile;
+  public Boolean hasTouch;
+  public String defaultBrowserType;
+
+  static DeviceDescriptor findByName(Playwright playwright, String name) {
+    JsonArray devices = ((PlaywrightImpl) playwright).deviceDescriptors();
```
Comment (uchagani) [inline playwright/src/main/java/com/microsoft/playwright/junit/impl/DeviceDescriptor.java:20]: @yury-s  this is returning null for me and so a null pointer exception is being thrown.

Comment (uchagani) [reply playwright/src/main/java/com/microsoft/playwright/junit/impl/DeviceDescriptor.java:20]: image

Comment (yury-s) [reply playwright/src/main/java/com/microsoft/playwright/junit/impl/DeviceDescriptor.java:20]: Hmm, that's odd. Is `options` in `options.deviceName` null or where does it come from?

Comment (uchagani) [reply playwright/src/main/java/com/microsoft/playwright/junit/impl/DeviceDescriptor.java:20]: I believe `PlaywrightImpl.deviceDescriptors()` is returning null so the for loop throws a null pointer exception:

image

Comment (yury-s) [reply playwright/src/main/java/com/microsoft/playwright/junit/impl/DeviceDescriptor.java:20]: Hmm, do you have a test where this is happening? This should never happen if playwright is initialized.

Comment (yury-s) [reply playwright/src/main/java/com/microsoft/playwright/junit/impl/DeviceDescriptor.java:20]: Ok, I assume it is just the new test. It passes on the bots and locally. Could it be that you have some stale driver? We changed the way device descriptors pushed to the client some time ago. Try downloading [current driver](https://github.com/microsoft/playwright-java/blob/main/CONTRIBUTING.md#updating-driver-version) and see if it still fails.

Comment (yury-s) [reply playwright/src/main/java/com/microsoft/playwright/junit/impl/DeviceDescriptor.java:20]: Just running `./scripts/download_driver_for_all_platforms.sh -f` and rebuilding should be sufficient

Comment (uchagani) [reply playwright/src/main/java/com/microsoft/playwright/junit/impl/DeviceDescriptor.java:20]: it was failing on main for me too but you figured it out.  downloaded the current driver and it's all good.  apologies for that.  i'll make running that command part of my workflow.

**Pull request merged**